ironfish/src/rpc/routes/peer/addPeerCandidate.ts

```diff
@@ -0,0 +1,51 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+import * as yup from 'yup'
+import { DEFAULT_WEBSOCKET_PORT } from '../../../fileStores/config'
+import { ApiNamespace, router } from '../router'
+
+export type AddPeerCandidateRequest = {
+  host: string
+  port?: number
+  whitelist?: boolean
+}
+
+export type AddPeerCandidateResponse = {
+  added: boolean
+  error?: string
+}
+
+export const AddPeerCandidateRequestSchema: yup.ObjectSchema<AddPeerCandidateRequest> = yup
+  .object({
+    host: yup.string().defined(),
+    port: yup.number().optional(),
+    whitelist: yup.boolean().optional(),
+  })
+  .defined()
+
+export const AddPeerCandidateResponseSchema: yup.ObjectSchema<AddPeerCandidateResponse> = yup
+  .object({
+    added: yup.boolean().defined(),
+    error: yup.string().optional(),
+  })
+  .defined()
+
+router.register<typeof AddPeerCandidateRequestSchema, AddPeerCandidateResponse>(
+  `${ApiNamespace.peer}/addCandidate`,
+  AddPeerCandidateRequestSchema,
+  (request, node): void => {
+    const peerManager = node.peerNetwork.peerManager
+    const { host, port, whitelist } = request.data
+
+    const peer = peerManager.getOrCreatePeer(null)
+    peer.setWebSocketAddress(host, port || DEFAULT_WEBSOCKET_PORT)
+    peer.isWhitelisted = !!whitelist
+
+    peerManager.peerCandidates.addFromPeer(peer)
+
+    const added = peerManager.connectToWebSocket(peer)
```

Comment:
This is pretty much the same as peerManager.connectToWebSocketAddress(), so I'd rather use that instead since the individual operations here are pretty important. Looks like we just need to pass through an option to force connecting even if we're above targetPeers?

Reply:
Ok agreed to pass down a parameter for forceConnect